pages/_app/index.jsx

```diff
@@ -178,18 +200,45 @@ class App extends NextApp {
     return null
   }
 
+  Login = () => {
```

Comment:
This looks weird, if you want this to be a component move this out of the _App_ and pass props, an `onLogIn` callback or whatever you need.

Reply:
I saw it used quite a lot in react projects. The reason why I used it here is that it is granted that iframe ref is assigned before componentDidMount is called. If it is extracted ref may not be assigned.

Reply:
Saying that I was assuming that you move ref into the Login component and handle all login process inside exposing only simple events.